---
title: Samtools
categories: 
- code
tags: 
- markdown
---

Samtools website: [samtools](http://samtools.sourceforge.net/samtools.shtml)

## View
Transfer `.sam` to `.bam` 
(`sort` can be only used on `.bam` instead of `.bam`)

`.bam`: binary file 
	1. Occupied less space than `.sam`
	2. Faster computing

Usage:

    samtools view [options] <in.bam>|<in.sam> [region1 [...]]
    
    Options:    -b  output BAM
                -h  print header for the SAM output
                -H  print header only (no alignments)
                -S  input is SAM
                -o  FILE  output file name [stdout]
                -f  INT   required flag, 0 for unset [0]
                -F  INT   filtering flag, 0 for unset [0]
                    4: The sequence is not aligned to the reference sequence
                    8: The mate sequence is not aligned to the reference sequence
                
Example:

    1. Transfer .sam to .bam
        $ samtools view -bS abc.sam > abc.bam
        $ samtools view -b -S abc.sam -o abc.bam
    2. Extract alignment result which align to the reference sequence
        $ samtools view -bF 4 abc.bam > abc.F.bam
    3. Extract the two reads in paired reads and compare them to the reference sequence. (12 = 4 + 8)
        $ samtools view -bF 12 abc.bam > abc.F12.bam

## sort
Usage:

    samtools sort [-n] [-m <maxMem>] <in.bam> <out.prefix>
    
    -m By default, the parameter is 500,000,000 or 500M.
    -n Sets the sorting method according to the ID of the short reads.
    
Example:

    $ samtools sort abc.bam abc.sort
    $ samtools view abc.sort.bam | less -S

## merge
The merged file does not need to be sorted.

Usage:

    samtools merge [-nr] [-h inh.sam] <out.bam> <in1.bam> <in2.bam>[...]
    
    Option: -n  sort by read names 
            -r  attach RG tag (inferred from file names)
            -u  uncompressed BAM output
    Note: Samtools' merge does not reconstruct the @RG dictionary in the header. Users must provide the correct header with -h, or uses Picard which properly maintains the header dictionary in merging.

<span style="color:red">*what does 'read names' mean from option -n?*</span>.
    
## Index
Index should be done after sorting.
Can generate `.bai` file which could be used in `samtools tview`, `gbrowse2`, and `IGV`.

Usage:

    samtools index <in.bam> [out.index]
    
Example: (same results in two ways)

    $ samtools index abc.sort.bam
$ samtools index abc.sort.bam abc.sort.bam.bai

## Faidx
Index the fasta file, generate `.fai` index files

Usage:

    samtools faidx <in.fasta> [ [...]]
    
    Index file in.fasta.fai is a text file
    Helps quickly extract sub-sequences in the fasta format from the genome
    
<span style="color:red">*what does 'sub-sequence' mean ?*</span>.

## tview
Intuitively showing the situation of reads versus the genome

Usage:

    samtools tview <aln.bam> [ref.fasta]
    
    1. First line: ref
    2. press 'g' plus a site number: go to the site
    3. H and backspace(left), J(up), K(down), L and space(right)
    4. Upcase: fast move; lowcase: slow move
    5. Ctrl+H: move left for 1kb base site; Ctrl+L: move right for 1kb base site
    6. Color mark:
        a. white: 30~40
        b. yellow: 20~30
        c. green: 10~20
        d. blue: 0~10

## Depth

Usage:

    bam2depth [-r reg] [-q baseQthres] [-Q mapQthres] [-b in.bed] <in1.bam> [...]
    
## [Bam to Fastq](http://www.hudsonalpha.org/gsl/information/software/bam2fastq)

## mpileup
Generate `.bcf` and then use `bcftools` to analyze SNP and Indel

## Bcftools
    
    
Reference:
1. [samtools常用命令详解](http://www.chenlianfu.com/?p=1399)
        
