---
title: CRISPR/CAS9 Gene Editing
categories: 
- biology
tags: 
- CRISPR
---

[Molecular biology at the cutting edge: A review on CRISPR/CAS9 gene editing for undergraduates](https://www.ncbi.nlm.nih.gov/pubmed/29381252)

# CRISPR: A Bacterial and Archeal Immune System Adapted for Eukaryotic Gene Editing

## Adaptive Immunity in Bacteria and Archea
![CRISPR/Cas9 mediated acquired immunity in prokaryotes](https://wol-prod-cdn.literatumonline.com/cms/attachment/1e47445c-4cfe-456b-8528-13609e98958d/bmb21108-fig-0001-m.jpg)
**Figure 1: CRISPR/Cas9 mediated acquired immunity in prokaryotes.** 
During the acquisition phase (A), cellular invaders such as phage virus inject nucleic acid sequences into the host cell. After infection, novel DNA sequences from the cellular invaders are incorporated into the host CRIPSPR locus as spacers (colored circles) flanked by repeat sequences (gray diamonds). As a result, when the CRISPR locus is transcribed, the pre‐CRISPR RNAs (crRNAs) encode the newly acquired protospacer sequences. The pre‐crRNA is cleaved to produce individual crRNAs that will associate with Cas proteins. The Cas protein utilizes the crRNAs as guides to silence foreign DNA that matches the crRNA sequence (B, interference phase). As a result, the second time a bacteria encounters the same foreign DNA, the crRNA/Cas9 complex is able to identify and silence the DNA.

1. CRISPR: Clustered Regulatory Interspaced Short Palindromic Repeats - a defense mechanism in an adaptive immune system (along with the CRISPR‐associated Proteins or Cas proteins)
2. By integrating DNA sequences that are identical to past invaders into their genome, bacteria and archea generate a cellular memory of past invaders. 
3. These acquired sequences allow the bacteria or archea to recognize viral or plasmid invaders as non‐self, resulting in the degradation of the invading sequence and functioning as an adaptive immune system for prokaryotes.
4. pre-crRNA/crRNA: CRISPR RNA
5. tracrRNA: trans‐activating CRISPR RNA
6. Double strand RNA = pre-crRNA + tracrRNA, cleaved by RNaseIII and then generate crRNA:tracrRNA complex containing just one spacer sequence - This RNA complex then associates with a single Cas9 protein, creating an active ribonucleoprotein (RNP) complex
5. Three different types of CRISPR–Cas systems have been identified in bacteria and archea: Type I, Type II, and Type III - Each system utilizes a different mechanism to generate crRNA and Cas proteins that catalyze the nucleic acid cleavage
6. Type II CRISPR system: most commonly adapted for genome editing due to its simplicity requiring just one Cas protein, Cas9, and two RNA components
7. PAMs: Proto‐spacer Adjacent Motifs
8. Cas9 nuclease domains used for cleavage: HNH (complementary DNA strand) and RuvC‐like (not complementary)
9. NHEJ: non‐homologous end joining
10. HDR: homology‐directed repair

![Figure 2](https://wol-prod-cdn.literatumonline.com/cms/attachment/58766ca1-e833-48b6-96bb-ed6c3d56a40d/bmb21108-fig-0002-m.jpg)
**Figure 2: Cas9 induced double‐strand breaks can be repaired by both nonhomologous end‐joining (NHEJ) or homology‐directed repair (HDR).**
(A) sequence of a targeted genomic locus in relation to the PAM (5′‐NGG‐3′) site. 
(B) Cartoon representation of crRNA, tracrRNA, and Cas9 protein assembly. 
(C) NHEJ results in random insertions, deletions, and indels. 
(D) HDR results in precise researcher‐designed edits. To achieve HDR, the researcher also introduces a repair template that contains the desired edit in which the HDR repair machinery of the cell uses to repair the induced double strand break.
  
![Figure 3](https://wol-prod-cdn.literatumonline.com/cms/attachment/76f2db09-8247-4a14-a94d-a3c7c2a23a0f/bmb21108-fig-0003-m.jpg)
**Figure 3: Cas9 has two nuclease domains each cutting a different strand of DNA.**
(A) Wildtype Cas9 contains two nuclease domains, RuvC and HNH which each cut a different strand of the DNA. When the RuvC nuclease domain is mutated, Cas9 will act as a nickase and produce a nicked DNA product (B).

## The Power of Making Programmed Double‐Strand Breaks for Genome Editing in Eukaryotes

1. C-NHEJ: canonical NHEJ - ligates or essentially “glues” the broken ends back together
2. alt‐NHEJ: alternative end joining pathway - one strand of the DNA on either side of the break is resected to repair the lesion
3. C-NHEJ and alt-NHEJ: both are error‐prone
